packages/predicates/src/is-noyes/index.bench.ts

```diff
@@ -0,0 +1,326 @@
+/*
+ * Copyright 2024 Hypergiant Galactic Systems Inc. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { bench, describe } from 'vitest';
+
+// biome-ignore lint/suspicious/noExplicitAny: <explanation>
+type Callable = (...a: any[]) => any;
+
+const TEST_VALUES = [
+  1,
+  0,
+  '1',
+  '0',
+  'on',
+  'off',
+  'true',
+  'false',
+  'yes',
+  true,
+  false,
+  'ON',
+  'YES',
+  'TRUE',
+] as const;
+
+const runTestValues = (fn: Callable) => TEST_VALUES.forEach(fn);
+
+// NOTE: doing a bench() inside of the for/of loop causes heap allocation errors
+describe('variations on isFalse|isNo|isOff|isOn|isTrue|isYes implementation', () => {
+  /**/ // <--- remove space to disable block comment
+  bench('original (regex) with setup', () => {
+    const trueRegex = /^(?:y|yes|true|1|on)$/i;
+    const falseRegex = /^(?:n|no|false|0|off)$/i;
+
+    function isFalse(val: unknown) {
+      const normalized = normalize(val);
+      return falseRegex.test(normalized);
+    }
+
+    function isTrue(val: unknown) {
+      const normalized = normalize(val);
+      return trueRegex.test(normalized);
+    }
+
+    function normalize(val: unknown) {
+      return `${val}`.trim();
+    }
+
+    runTestValues(isTrue);
+    runTestValues(isFalse);
+  });
+  //*/
+
+  /** / // <--- remove space to disable block comment
+  (() => {
+    // isolated scope
+    const trueRegex = /^(?:y|yes|true|1|on)$/i;
+    const falseRegex = /^(?:n|no|false|0|off)$/i;
+
+    function isFalse(val: unknown) {
+      const normalized = normalize(val);
+      return falseRegex.test(normalized);
+    }
+
+    function isTrue(val: unknown) {
+      const normalized = normalize(val);
+      return trueRegex.test(normalized);
+    }
+
+    function normalize(val: unknown) {
+      return `${val}`.trim();
+    }
+
+    bench('original (regex) without setup', () => {
+      runTestValues(isTrue);
+      runTestValues(isFalse);
+    });
+  })();
+  //*/
+
+  /**/
+  bench('[altered] original (regex) with setup', () => {
+    const trueRegex = /^(?:y|yes|true|1|on)$/i;
+    const falseRegex = /^(?:n|no|false|0|off)$/i;
+
+    const test = (regex: RegExp, val: unknown) => regex.test(`${val}`.trim());
```

Comment:
We may want to hoist this function and potentially the regex outside of `bench` to remove any overhead from function allocation etc (i'm not sure if values created in a `bench` callback are created each run of of the bench)

Reply:
More than likely are, third "best case" scenario that I linked above. (Though it is from a different platform)

Reply:
Yea, I find "setup" to still be partially relevant since it does change start-up times. I usually run two benchmarks side-by-side to make sure that it isn't egregious. But I also don't know how we want to do it in this repo.